src/main/java/com/hillel/items_exchange/security/jwt/JwtTokenProvider.java

```diff
@@ -87,4 +89,19 @@ public boolean validateToken(String token, HttpServletRequest req) {
             return false;
         }
     }
+
+    public void invalidateToken(String token) {
+        final Date expirationDate = getTokenExpirationDate(token)
+                .orElseThrow(() -> new JwtException(getExceptionMessageSource("invalid.token")));
+        invalidatedTokensHolder.invalidate(token, expirationDate);
+    }
+
+    public Optional<Date> getTokenExpirationDate(String token) {
+        try {
+            return Optional.of(Jwts.parser().setSigningKey(secret).parseClaimsJws(token).getBody().getExpiration());
```

Comment:
I just want to clearify that Optiona.of doesn't expect null value. Did you aware of it when created the code?